soh/soh/Enhancements/randomizer/randomizer_entrance.c

```diff
@@ -586,25 +671,73 @@ void Entrance_OverrideGeurdoGuardCapture(void) {
 }
 
 void Entrance_OverrideSpawnScene(s32 sceneNum, s32 spawn) {
+    // Copy the actorentry properties to avoid modifying the original cached pointer
+    // The assign a pointer of our modified actoreEntry back
```

Comment:
why is this required?

Reply:
Scene resources are cached, so any modification to the pointer is permanently retained even for vanilla, and we can't invalidate scenes currently (my same compliant w.r.t. water box collisions also modifying cached scene resources). 

Since playstate just holds actorentry as a pointer and doesn't clear it, I can have this object here and just replace the values as needed, without modifying the original pointer.